### PR TITLE
fix: dns interceptor ip ttl

### PR DESCRIPTION
**Pull request by luddd3**
## This relates to...

How long time to cache the dns records from the dns interceptor.

## Rationale

For how long time the dns records from the dns interceptor should live.

## Changes

Set timestamp once on each resolved record and compare the ttl to that timestamp. If the record is too old, remove it.

### Bug Fixes

N/A

## Status

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [x] Tested
- [S] Benchmarked (**optional**)
- [S] Documented
- [x] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: lib/interceptor/dns.js [hunk 1]
```diff
@@ -171,24 +165,24 @@ class DNSInstance {
...
     }
 
-    ip.timestamp = timestamp
-
     this.lastIpFamily = newIpFamily
     return ip
   }
 
-  setRecords (origin, addresses) {
+  setRecords (origin, opts, addresses) {
```
Comment (metcoder95) [inline lib/interceptor/dns.js:179]: No need to pass the opts, the options are already attached to instance properties

File: lib/interceptor/dns.js [hunk 2]
```diff
@@ -171,24 +165,24 @@ class DNSInstance {
...
+    const timestamp = Date.now()
     const records = { records: { 4: null, 6: null } }
     for (const record of addresses) {
+      record.timestamp = timestamp
+      record.ttl = opts.maxTTL
```
Comment (metcoder95) [inline lib/interceptor/dns.js:184]: If the record already contains a `ttl` let's preserve it, otherwise default to `maxTTL`.

On top, let's use the min from both values.

Comment (luddd3) [reply lib/interceptor/dns.js]: I have added it.

How do you feel about expecting the record result to be in ms? `dns.lookup` doesn't usually provide ttl. `dns.resolve` does, but it's ttl, and other time properties, are in seconds.
https://nodejs.org/api/dns.html#dnsresolveanyhostname-callback

File: test/interceptors/dns.js [hunk 3]
```diff
@@ -546,31 +550,13 @@ test('Should we handle TTL (4)', async t => {
...
-                address: addr.address,
-                ttl: opts.maxTTL,
-                family: addr.family
-              }
-
-              results.set(`${record.address}:${record.family}`, record)
-            }
-
-            cb(null, results.values())
-          }
```
Comment (mcollina) [inline test/interceptors/dns.js:573]: why was this deleted?

Comment (luddd3) [reply test/interceptors/dns.js:573]: I didn't think it adds any value to the test. Does it?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by metcoder95 — APPROVED** [review]
